src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -589,31 +591,47 @@ void DirectLightingIntegrator::add_emitting_triangle_sample_contribution(
     double cos_on = dot(-incoming, sample.m_shading_normal);
     if (cos_on <= 0.0)
         return;
-
-    // Compute the transmission factor between the light sample and the shading point.
-    const float transmission =
-        m_shading_context.get_tracer().trace_between(
-            m_shading_point,
-            sample.m_point,
-            VisibilityFlags::ShadowRay);
-
-    // Discard occluded samples.
-    if (transmission == 0.0f)
-        return;
-
+    
     // Compute the square distance between the light sample and the shading point.
     const double square_distance = square_norm(incoming);
-    const double rcp_sample_square_distance = 1.0 / square_distance;
-    const double rcp_sample_distance = sqrt(rcp_sample_square_distance);
-
+    
     // Don't use this sample if we're closer than the light near start value.
     if (square_distance < square(edf->get_light_near_start()))
         return;
 
-    // Normalize the incoming direction.
-    incoming *= rcp_sample_distance;
+    const double rcp_sample_square_distance = 1.0 / square_distance;
+    const double rcp_sample_distance = sqrt(rcp_sample_square_distance);
+    
+    // Normalize the incoming direction. 
     cos_on *= rcp_sample_distance;
+    incoming *= rcp_sample_distance;
+    
+    // Decide whether the contributon from this light is significant or not.
+    const double approximate_contribution = 
+        cos_on * 
+        dot(incoming, m_geometric_normal) * 
```

Comment:
I'm really not sure it's a good idea to take into account `incoming` at all... But if you do, you should use the shading normal instead of the geometric normal.